fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java

```diff
@@ -1574,21 +1579,36 @@ public void validateAccountBalanceDoesNotBecomeNegative(final String transaction
                         throw new PlatformApiDataValidationException(dataValidationErrors);
                     }
                 }
-
             }
             lastSavingsDate = transaction.transactionLocalDate();
-
         }
 
         BigDecimal withdrawalFee = null;
-        BigDecimal transactionAmount = null;
         if (isOverdraft()) {
             if (runningBalance.minus(minRequiredBalance).isLessThanZero()) {
-                throw new InsufficientAccountBalanceException("transactionAmount", getAccountBalance(), withdrawalFee, transactionAmount);
+                overdraftLimitExceededException("transactionAmount", getAccountBalance(), withdrawalFee, transactionAmount);
             }
         }
     }
 
+    public void validateAccountBalanceDoesNotViolateOverdraft(final List<SavingsAccountTransaction> savingsAccountTransaction,
+            final BigDecimal amountPaid) {
+        if (savingsAccountTransaction != null) {
+            SavingsAccountTransaction savingsAccountTransactionFirst = savingsAccountTransaction.get(0);
+            if (!this.allowOverdraft) {
+                if (savingsAccountTransactionFirst.getRunningBalance(this.currency).minus(amountPaid).isLessThanZero()) {
+                    throw new InsufficientAccountBalanceException("chargeAmount", getAccountBalance(), null, amountPaid);
+                }
+            }
+        }
+    }
+
+    private void overdraftLimitExceededException(final String paramName, final BigDecimal accountBalance, final BigDecimal withdrawalFee,
+            final BigDecimal transactionAmount) {
+        String showMessageForLimitExaustion = "Overdraft Limit is Exceeded";
+        throw new PlatformDataIntegrityException(showMessageForLimitExaustion, showMessageForLimitExaustion);
```

Comment:
Is there a reason to use PlatformDataIntegrityException instead of InsufficientAccountBalanceException?
Also all of the passed parameters are ignored: paramName, accountBalance, withdrawalFee, transactionAmount. Can you please write them into log with WARN at least?

Reply:
Thanks for the review, @adamsaghy Comment addressed.